spec.bs

```diff
@@ -501,14 +506,59 @@ are:
     1. Let |permission| be the result of [=checking interest group permissions=] with
       |owner|, |frameOrigin|, and "`leave`".
     1. If |permission| is false, then [=queue a task=] to [=reject=] |p| with a
-       "{{NotAllowedError}}" {{DOMException}} and do not run the remaining steps.
+       "{{NotAllowedError}}" {{DOMException}} and abort these steps.
     1. [=Queue a task=] to [=resolve=] |p| with `undefined`.
     1. [=list/Remove=] [=interest groups=] from the user agent's [=interest group set=] whose
       [=interest group/owner=] is |owner| and [=interest group/name=] is |name|.
 1. Return |p|.
 
 </div>
 
+<h3 id="clearoriginjoinedAdInterestGroups">clearOriginJoinedAdInterestGroups()</h3>
+{{Window/navigator}}.{{Navigator/clearOriginJoinedAdInterestGroups()}} removes a user from
+[=interest groups=] whose [=interest group/joining origin=] is the current top level page's
+origin.

```

Comment:
What happens if this is called in a fenced frame? Why does "top level page" mean there? Basically I'm trying to see if we can link to one of these:
 - https://wicg.github.io/fenced-frame/#top-level-traversable
 - https://html.spec.whatwg.org/multipage/document-sequences.html#nav-traversable

so we can make this more rigorous.

Reply:
When run in a fenced frame, it behaves exactly like leaveAdInterestGroup() does when passed a non-empty group: It throws.  Note that I exactly copied the verbiage for that, so if one is wrong, they both are.  I assume the throwing is covered by the 'not allowed to use the "join-ad-interest-group" policy-controlled feature' bit, but could be wrong.

I copied "top level page's origin" from the definition of "joining origin" ("The top level page origin from where the interest group was joined.") - reusing terminology used when defining joining origin when mentioning joining origin seems reasonable to me.  If that terminology is wrong, then that definition needs to be updated as well.  The doc repeatedly refers to "top-level origin", which may be the term we want?  I've replaced it here, but left joining origin's definition alone.

Reply:
The text for setting joining origin in joinAdInterestGroup is: "Set interestGroup’s joining origin to this's relevant settings object's top-level origin."

Reply:
OK thanks, so the top-level origin is scoped to a fenced frame here, and does not transcend the boundary (i.e., does not "jump" the fence).

In that case, I think the test we should use here is:

```
...removes a user from [=interest groups=] whose [=interest group/joining origin=]
is the associated {{Navigator}}'s [=relevant settings object=]'s
[=environment/top-level origin=]
```

(Which should link to https://html.spec.whatwg.org/multipage/webappapis.html#relevant-settings-object and https://html.spec.whatwg.org/multipage/webappapis.html#concept-environment-top-level-origin respectively, of course)

Does that make sense / sound good?

Reply:
Sounds good, done.

Unless we're letting these fenced frames set 1P cookies, I suspect this behavior is not what we want, long term, since this is much akin to setting 1P cookies for the embedded frame (albeit 1P cookies that can only be accessed in an auction - using the top-level-site's joining origin are more like partitioning cookies), but don't think this PR is the place to iron that out.

Reply:
That's interesting. Let me just cc @shivanigithub just for extra exposure, but I agree that it shouldn't have a bearing on this PR specifically.